src/archives.rs

```diff
@@ -13,9 +13,11 @@ pub(crate) enum ArchiveFormat {
 impl ArchiveFormat {
     /// Parse archive type from resource extension.
     pub(crate) fn parse_from_extension(resource: &str) -> Result<Self, Error> {
-        if resource.ends_with(".tar.gz") {
+        let ext = infer::get_from_path(resource).unwrap().unwrap().extension();
+        // Here we assume that gz contain a tar inside it.
+        if ext.ends_with("gz") {
```

Comment:
I haven't used `infer` before, but do we need `.ends_with()` or can we just check for equality? Though we might to check for both "gz" and "tar.gz"...

Reply:
I think we can use equality here. `infer` for zip and gz just returns the string `zip` or `gz` without the dot. I was thinking about the tar.gz case, but infer can't guess that.